verilog/analysis/default_rules.h

```diff
@@ -24,6 +24,7 @@ namespace analysis {
 // LINT.IfChange
 constexpr const char* kDefaultRuleSet[] = {
     "invalid-system-task-function",
+    "instance-shadowing",
```

Comment:
Like all new rules first comment this out (see list at the bottom of this array). Once implemented, we can inform everyone on the verible mailing list that this will be enabled and then go from there.
That way, people can adapt to potentially newly triggering lint rules.